login/login.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package login
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/usso"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+)
+
+type TokenGetter interface {
+	GetTokenWithOTP(email, password, otp, tokenName string) (*usso.SSOData, error)
+}
+
+var ussoServer (TokenGetter) = usso.ProductionUbuntuSSOServer
+
+// LoginUSSO reads login parameters from ctxt.Stdin and then retrieves an
+// oauth token from Ubuntu SSO.
+func LoginUSSO(ctx *cmd.Context, twoFactor bool, store TokenStore) (*usso.SSOData, error) {
+	email, pass, otp, err := readUSSOParams(ctx, twoFactor)
+	if err != nil {
+		return nil, errgo.Notef(err, "cannot read login parameters")
+	}
+	tok, err := ussoServer.GetTokenWithOTP(email, pass, otp, "charm")
+	if err != nil {
+		return nil, errgo.Notef(err, "cannot get token")
+	}
+	if err := store.SaveToken(tok); err != nil {
```

Comment:
I'm not sure that this function should both return the token *and* save it
to the store. If we got rid of the store argument and just returned the token,
then I think the need for nopStore would go away.